File: bio-informatics/challenges/check-if-clump.js
const checkIfClump = (list, kMerLength, oriLength, occurences) => {
    for (let i = 0; i < list.length; i++) {
        const first = list[i];
        const last = list[i + occurences - 1];
        if( ( last + kMerLength - first ) <= oriLength) {
            return true;
        }
    }

    return false;
}

module.exports = checkIfClump;

File: bio-informatics/challenges/python/utils/check_if_clump.py
def check_if_clump(list, k_mer_length, ori_length, occurences):
  for x in range(len(list)):
    first = list[x]
    last = list[x + occurences - 1]
    if last + k_mer_length - first <= ori_length:
      return True
  return False
